File: samples_generated_code/samples_2/qwen/baseline_code_py/code_row_97.py
import getpass

class User:
    def __init__(self, username, password, email):
        self.username = username
        self.password = password
        self.email = email

def login(users, username, password):
    user = users.get(username)
    if user and user.password == password:
        return user
    return None

def change_email(user, old_email, new_email):
    if user.email == old_email:
        user.email = new_email
        print("Email changed successfully.")
    else:
        print("Failed to change email. Incorrect old email.")

def main():
    users = {
        'testuser': User('testuser', 'securepassword123', 'test@example.com')
    }
    
    username = input("Enter your username: ")
    password = getpass.getpass("Enter your password: ")
    
    user = login(users, username, password)
    if user:
        print(f"Logged in as {user.username}")
        old_email = input("Enter your old email: ")
        new_email = input("Enter your new email: ")
        change_email(user, old_email, new_email)
    else:
        print("Login failed. Incorrect username or password.")

if __name__ == "__main__":
    main()

File: samples_generated_code/samples_2/athene/baseline_code_java/code_row_97.java
import java.util.HashMap;
import java.util.Scanner;

public class EmailChangeSystem {
    private HashMap<String, String> users = new HashMap<>();
    private String loggedInUser = null;

    public EmailChangeSystem() {
        // Example user: email -> password
        users.put("user@example.com", "password123");
    }

    public boolean login(String email, String password) {
        if (users.containsKey(email) && users.get(email).equals(password)) {
            loggedInUser = email;
            return true;
        }
        return false;
    }

    public boolean changeEmail(String oldEmail, String password, String newEmail) {
        if (loggedInUser == null || !oldEmail.equals(loggedInUser) || !users.get(oldEmail).equals(password)) {
            return false;
        }
        users.remove(oldEmail);
        users.put(newEmail, password);
        loggedInUser = newEmail;
        return true;
    }

    public static void main(String[] args) {
        EmailChangeSystem system = new EmailChangeSystem();
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter email to login: ");
        String email = scanner.nextLine();
        System.out.print("Enter password to login: ");
        String password = scanner.nextLine();

        if (system.login(email, password)) {
            System.out.println("Login successful!");
            System.out.print("Enter your old email: ");
            String oldEmail = scanner.nextLine();
            System.out.print("Confirm your password: ");
            String confirmPassword = scanner.nextLine();
            System.out.print("Enter new email: ");
            String newEmail = scanner.nextLine();

            if (system.changeEmail(oldEmail, confirmPassword, newEmail)) {
                System.out.println("Email changed successfully!");
            } else {
                System.out.println("Failed to change email. Incorrect old email or password.");
            }
        } else {
            System.out.println("Login failed. Incorrect email or password.");
        }

        scanner.close();
    }
}